### DISCUSSION TITLE
Suggestions on how to implement point by point additional task during scanning

### DISCUSSION THREAD

**Discussion by malobezard** [Ulm-IQO/qudi-core#87] [Q&A] [answered]
Hello, 

Withint the framework of my experiments, I am trying to find a way to sequentially turn on & off my cw laser while performing a confocal scan. As an example, this would typically mean being able, at each point of the scan to
- turn on the laser for e.g. 100 ms for acquisition
- turn off the laser & move the scanners after a 10ms delay 

(Note that the delay could be placed before of after the aquisition period)

It was already pointed out to me that there exists a "sample_clock_output" parameter in the scanning toolchain allowing for associating a physical digitial channel to the scanner clock. In that way, I could output a 5V voltage output synchronously with the scanning procedure. However, this does not cover the duty cycle issue. 

Would you have any ideas/insights on how to perform/implement such a task ?

Thank you

Malo

**Comment by LukePiewalker** [comment, marked as answer]
Dear @malobezard,

sorry for the late reply.
To change the duty cycle for the sample clock you just need to pass an additional argument, as outlined here
[nidaqmx docu: COChannelCollection.add_co_pulse_chan_freq](https://nidaqmx-python.readthedocs.io/en/latest/co_channel_collection.html#nidaqmx._task_modules.co_channel_collection.COChannelCollection.add_co_pulse_chan_freq), upon creation of the sample clock task in the [NIXSeriesFiniteSamplingIO, line 781](https://github.com/Ulm-IQO/qudi-iqo-modules/blob/6cd2ff89143e6d45f2c5b35c09c91d2d73ab9818/src/qudi/hardware/ni_x_series/ni_x_series_finite_sampling_io.py#L780).
I did not try this myself with real hardware thus far, so please give feedback if it worked as expected.
